lib/duniverse.mli

```diff
@@ -1,13 +1,87 @@
-type t = {
-  root_packages : Types.Opam.package list;
-  excludes : Types.Opam.package list;
-  pins : Types.Opam.pin list;
-  packages : Types.Opam.entry list;
-  repos : Types.Dune.repo list;
-  remotes : string list; [@default []]
-  branch : string [@default "master"]
-}
-[@@deriving sexp]
+module Deps : sig
+  module Opam : sig
+    (** Type of dependencies to install through opam *)
+    type t = { name : string; version : string option }
+
+    val equal : t -> t -> bool
+
+    val pp : t Fmt.t
+  end
+
+  module Source : sig
+    (** Type of dependencies to clone in the duniverse *)
+    type t = { dir : string; upstream : string; ref : string; provided_packages : Opam.t list }
+
+    val equal : t -> t -> bool
+
+    (**/**)
+
+    (* Exposed for test purposes only *)
+
+    val raw_pp : t Fmt.t
```

Comment:
`raw_pp` because it formats using ocaml syntax? Writing printers by hand seems cumbersome. Using `[@@deriving sexp]` is much easier and avoid adding bugs.

Reply:
trying to avoid using ppx here I think -- we can't when its merged into dune

Reply:
Yep we don't want to use PPX-es anymore ahead of a dune merge.